modules/layers/src/base-tile-layer/utils/tile-cache.js

```diff
@@ -1,5 +1,33 @@
 import Tile from './tile';
 
+const TILE_STATE_UNKNOWN = 0;
+const TILE_STATE_VISIBLE = 1;
+/*
+   show cached parent tile if children are loading
+   +-----------+       +-----+            +-----+-----+
+   |           |       |     |            |     |     |
+   |           |       |     |            |     |     |
+   |           |  -->  +-----+-----+  ->  +-----+-----+
+   |           |             |     |      |     |     |
+   |           |             |     |      |     |     |
+   +-----------+             +-----+      +-----+-----+
+
+   show cached children tiles when parent is loading
+   +-------+----       +------------
+   |       |           |
+   |       |           |
+   |       |           |
+   +-------+----  -->  |
+   |       |           |
+ */
+const TILE_STATE_PLACEHOLDER = 3;
+const TILE_STATE_HIDDEN = 4;
+// tiles that should be displayed in the current viewport
+const TILE_STATE_SELECTED = 5;
+
+export const STRATEGY_EXCLUSIVE = 'exclusive';
+export const STRATEGY_DEFAULT = 'default';
+
 /**
```

Comment:
This file looks quite complicated, and happens to be a piece where Unfolded needs to do a lot of work. It would be good to see if we can improve the structure a bit. 

As my initial reaction, I would prefer aligning this code with the class structure in the 3D tiles implementation, namely:

- A `Tileset2D` class that accepts a viewport and return a list of loaded tiles to display, and also accepts callbacks to call when new tiles load. 
- Thus the TileLayer would take a `Tileset2D` instance and we could move most differences between various tile loading strategies out of the `Tileset2D` layer.
- A separate `Tile2DCache` class that handles any actual caching.

I am not yet 100% sure if I am mainly just proposing renaming this class, or if there is meaningful simpliciation/decomposition that can be done this way.